components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/importexport/utils/APIAndAPIProductCommonUtil.java

```diff
@@ -0,0 +1,650 @@
+                        IOUtils.copy(imageDataStream, outputStream);
+                        if (log.isDebugEnabled()) {
+                            log.debug("Thumbnail image retrieved successfully for API/API Product: " + identifier.getName()
+                                    + StringUtils.SPACE + APIConstants.API_DATA_VERSION + ": "
+                                    + identifier.getVersion());
+                        }
+                    }
+                } else {
+                    //api gets imported without thumbnail
+                    log.error("Unsupported media type for icon " + mediaType + ". Skipping thumbnail export.");
+                }
+            } else if (log.isDebugEnabled()) {
+                log.debug("Thumbnail URL [" + thumbnailUrl + "] does not exists in registry for API/API Product: "
+                        + identifier.getName() + StringUtils.SPACE + APIConstants.API_DATA_VERSION + ": "
+                        + identifier.getVersion() + ". Skipping thumbnail export.");
+            }
+        } catch (RegistryException e) {
+            log.error("Error while retrieving API/API Product Thumbnail " + thumbnailUrl, e);
+        } catch (IOException e) {
+            //Exception is ignored by logging due to the reason that Thumbnail is not essential for
+            //an API to be recreated.
+            log.error("I/O error while writing API/API Product Thumbnail: " + thumbnailUrl + " to file", e);
+        }
+    }
+
+    /**
+     * Retrieve documentation for the exporting API or API Product and store it in the archive directory.
+     * FILE, INLINE, MARKDOWN and URL documentations are handled.
+     *
+     * @param identifier    ID of the requesting API or API Product
+     * @param registry      Current tenant registry
+     * @param docList       Documentation list of the exporting API or API Product
+     * @param exportFormat  Format for export
+     * @throws APIImportExportException If an error occurs while retrieving documents from the
+     *                                  registry or storing in the archive directory
+     */
+    public static void exportAPIOrAPIProductDocumentation(String archivePath, List<Documentation> docList,
+                                               Identifier identifier, Registry registry, ExportFormat exportFormat)
+            throws APIImportExportException {
+
+        Gson gson = new GsonBuilder().setPrettyPrinting().create();
+        String docDirectoryPath = File.separator + APIImportExportConstants.DOCUMENT_DIRECTORY;
+        CommonUtil.createDirectory(archivePath + docDirectoryPath);
+        try {
+            for (Documentation doc : docList) {
+                String sourceType = doc.getSourceType().name();
+                String resourcePath = null;
+                String localFilePath;
+                String localFileName = null;
+                String localDocDirectoryPath = docDirectoryPath;
```

Comment:
docDirectoryPath is not used further. So do we need a new variable? Can we reuse docDirectoryPath?

Reply:
Yes we can resue it. Thanks for pointing it out.